### ISSUE TITLE
Statistics on Root Forms and Root Languages

### ISSUE THREAD

**Issue by CasFre** [lingdb/CoBL#236]
Hello Jakob,
Would it be possible that you compute some statistics for me on:
- How many cognate sets are there that contain lexemes from more than _two branches_, but do not have a root form yet? And which cognate sets are these?
- How many cognate sets are there that contain lexemes from more than _one branch_, but do not have a root form yet? And which cognate sets are these?

My impression is that those classes which contain more than two branches and are "good Indo-European" are well treated by now, but I would like to check in preparation for our envisaged meeting on Tuesday.

I am not sure whether from your perspective this issue should be merged with #225 .

**Comment by runjak** [comment]
Hi @CasFre thanks for putting a separate issue for this!
I prefer smaller issues because it makes for more concise discussions
and it's simpler to figure out when they can be closed.

I'll provide the cognate classes for the `Current` language list and the `Jena200` meaning list.

**Comment by runjak** [comment]
Another question:
When you write *branches* - do you refer the the clades or to the languages in the database?

**Comment by PaulHeggarty** [comment]
Not sure this really deserves priority 0 right now, with so much else desperately needed. Jakob, finish off some other stuff first.
By the way, Cassandra, you could at least focus down on things without a root form by typing ^$ in the root form filter box.  That would help you look across to the colour bars to see which ones have more than one colour bar...

**Comment by runjak** [comment]
I was already working on this and it made sense to get it finished.

@CasFre this could be useful:

File: | Entries:
----- | -----
[task1.txt](https://github.com/lingdb/CoBL/files/378312/task1.txt) | 96
[task2.txt](https://github.com/lingdb/CoBL/files/378313/task2.txt) | 363

**Comment by CasFre** [comment]
@runjak: Yes, I meant clades such as Anatolian, Germanic, Indo-Iranian.

@PaulHeggarty: Typing ^$ unfortunately does not help much with most meanings where you have loads of clade-individual cognate classes which tend not to have root forms yet.
Of course, Jakob surely has important programming work to do, but I would say that my issue nevertheless is important as well because at this stage of the project I would like to get an overview of how the cognacy coding has been going which is not just based on my intuitions, but on facts and which would help me focus on those issues where my work really can make a significant change.
As you said during our last catchup, I should definitely focus on those cognate classes that contain lexemes from more than one clade/branch, which I have been doing since then, but my impression was when browsing through the meanings that most of that work seems to have been done already.
So for me the question arises whether there still are gaps in this sort of coding, or, if there aren't any (or any solvable ones), whether I should focus on other things again, e.g. cross-checking cognacy coding with those classes that involve lexemes from only one branch, but might have been erroneously put up as a separate cognate class instead of being linked to an existing bigger one containing other branches, or tidying up references to older handbooks before the launch which might be a job to do before the launch and to be shared between Kontantin and me.
So I would say that, although maybe it's not a 0, it is definitely something worth doing in the course of the next few days in order to provide a structure for the work on the cognacy coding which is a vital part of CoBL as well.

**Comment by CasFre** [comment]
Thank you @runjak. So these are the lists with the parameters that I asked for?
Could you easily add the meaning that these cognate classes belong to? - I forgot that it is as of yet not possible to navigate to individual cognate classes directly.

**Comment by runjak** [comment]
Sure thing - adding the meaning will not be too complicated.
I'll have a bit delay since I'm currently tackling a different problem, but I'll get to it eventually.

**Comment by runjak** [comment]
Yes, the lists are for the given parameters with `task1.txt` corresponding to the first one.
I've redone the lists so the individual entries are a bit more verbose and include the meanings a cognate class belongs to:

File | Entries
----- | -----
[task1.txt](https://github.com/lingdb/CoBL/files/379454/task1.txt) | 92
[task2.txt](https://github.com/lingdb/CoBL/files/379455/task2.txt) | 360

**Comment by CasFre** [comment]
Thanks @runjak for the redoing.

Quick report for myself and all others (esp. @m-scarborough, @nitsnatsnok):

- overall number of cognate sets in CoBL: 5715 (by 2016-07-25, 09:37 am)
- 120 cognate sets with lexemes from 3+x branches which weren’t assigned a root form yet
- 301 cognate sets with lexemes from exactly two branches which weren’t assigned a root form yet ([Σ CogCl 2+x] – [Σ CogCl 3+x] = 421 – 120 = 301), of these, by estimate(!), 1/2 or 3/5 are inner-Indo-Iranian

So I am going to focus on the cognate sets from these lists for now, starting with the 3+x first; working them through and filling in root forms, or – where not possible – leave some notes for later revision. I am going to upload the working report here on Github unless somebody suggests another, better way of dealing with that.

**Comment by runjak** [comment]
Cool!
I'll reassign this issue to @CasFre then and consider the programming part as solved for now :)

**Comment by PaulHeggarty** [comment]
Excellent.  Now that I see the actual results from these statistics, I see that you were dead right about the priority rank, @CasFre.  Good call!

**Comment by CasFre** [comment]
Thank you for that, @PaulHeggarty!

**Comment by PaulHeggarty** [comment]
What would also be nice as a progress tracker is a total count of all cognate sets
- across all Jena 200 meanings 
- for the Current languages list
- filtered to exclude cognate sets which have only 'Not Swh' marked lexemes
- and where the cognate set has no root form specified.

The same also for the above combination but where the last criterion is instead:
- and where no root language is specified.

**Comment by CasFre** [comment]
Hello @runjak,
http://www.cobl.info/cognate/974/ also was on the list, although I assigned a root form and root language to it some months ago.
As things like this might give wrong figures for further statistics: What might be the reason for that and can we eliminate it?

**Comment by runjak** [comment]
I believe the reason for `974` turning up falsely could have been that I'm computing the statistics on my machine with a state of the database that doesn't change synchronized with the database on the server. If we integrate these statistics directly into the website so that they can be requested there such a problem should no longer turn up.

**Comment by CasFre** [comment]
I finished the list with those cognate classes that contain lexemes from at least three branches and hadn't been assigned a root form.
I upload my notes here: 
[task1_incl meanings_CF.docx](https://github.com/lingdb/CoBL/files/400184/task1_incl.meanings_CF.docx)
There were some inner-Indo-Iranian cases as well as a substantial amount of inner-Germanic and inner-Celtic cognate sets among them which I as a general policy did not tackle yet, leaving the Indo-Iranian cases to @m-scarborough and @RolandPooth, and the Celtic ones to @cormacanderson.
You should easily find them in the document by opening a search field and typing "inner-Indo-Iranian" or "inner-Celtic" respectively.

**Comment by PaulHeggarty** [comment]
Since there are 19 inner-Germanic cognate sets, noted as 'to come back to later', would you feel confident enough in Proto-Germanic, Cassandra, to take those on yourself?

**Comment by CasFre** [comment]
As we don't have a Germanic specialist at hand, but rather decent reference works with Kroonen's dictionary and the published parts of Etymologisches Wörterbuch des Althochdeutschen, I think it should be a good idea that I have a look at them and see how much of them I can solve within a reasonable amount of time.

**Comment by m-scarborough** [comment]
Ok, I think I misunderstood the task assigned to me in the previous emails, since I've been going through the task1.txt list also (not task2-task1).  My apologies, everyone.  That was my mistake.  The upshot of this is that I've checked the cognacy on quite a few of the ones marked 'finished: n' and have input root forms and root languages on quite a few of them (including some of those Germanic ones where there are well established PIE roots in LIV).  I'll have a run through and update your file to specify my changes.

**Comment by PaulHeggarty** [comment]
It certainly doesn’t sound like you have anything to apologise for, in any case!  This sounds like good news, that things are already a bit further advanced than the file says.  Please do update it and re-post here.

**Comment by PaulHeggarty** [comment]
Would it be useful to have a 'Revised yet?' field for each cognate class?

**Comment by m-scarborough** [comment]
That could be helpful, as that would help spare us these word documents.

Actually, I wonder whether it might be possible to have that, and a page with sorting and filters like the 1-meaning in all cognate sets pages, but listing all cognate sets 1-6018 (or whatever we're up to now).  That would save us time tremendously from having to keep all this information in separate word or .txt files that we update manually.  If that's not something we can do due to programming priorities at the moment, can it be something to be implemented later on?

**Comment by PaulHeggarty** [comment]
Funny, that was exactly what was keeping me awake last night.  We definitely do need a list of all cognate sets across the entire project.  The big potential problem is that cognate classes are determined by meaning, so the sorting and filters may be really tricky.  I’ll see what Jakob thinks today.  
We could also do with a field not just for 'Revised yet?' but also who is the 'Reviser'.
This is part of the eternal problem of balancing the priorities for programming new functions for the cognate coders on the one hand, and for the language data entry people on the other hand.  It's a very intense phase for both right now.

**Comment by PaulHeggarty** [comment]
I've converted the Word document, as revised by Matthew, into a page in the wiki:

https://github.com/lingdb/CoBL/wiki/08--Multi-Branch-Cognate-Sets

This should be a bit faster to use, because I’ve inserted direct hyperlinks to the cognate sets referred to.  Perhaps more useful like this, so that you can work by editing the wiki now, rather than the word document, and it will be easier to keep updated for everyone to see.

**Comment by PaulHeggarty** [comment]
I’ve just updated, renamed and relinked the wiki page to:
https://github.com/lingdb/CoBL/wiki/08--Cognate-Sets-used-across-3-or-more-Branches  

This is because we should probably have a second page for cognate sets used in exactly two branches.  However, we are in any case going to re-do the statistics now, and get new results posted here, hopefully later today.  

Cassandra, can you let us know what stage you are at with task 2?  For example, have you already prepared a 'with meanings' word document for it, or is the latest version just task2 minus task1.docx?

**Comment by CasFre** [comment]
Hello Paul,
the "task 2 minus task 1.docx" already contains the meanings. I derived it from Jakob's updated statistics version including meanings.
I did not work on it yet myself, because I had assigned it to Matthew - at least as far as the Indo-Iranian cases are concerned- and I also sent Roland Pooth the link to it yesterday so that he may also have a look at it from the Indic side of it. But I think the Vedic lists remain his priority for the moment?
So you may feel free to re-generate a new list of those cognate sets with lexemes from 2 clades in it, but no root forms + root language.

**Comment by CasFre** [comment]
By the way, thanks for putting up the wiki page. For the moment, this is certainly a good solution.

**Comment by runjak** [comment]
Paul asked me to provide the statistics again:

File | Entries
----- | -----
[1.txt](https://github.com/lingdb/CoBL/files/401609/1.txt) | 39
[2.txt](https://github.com/lingdb/CoBL/files/401610/2.txt) | 303

I'll provide a different version in a bit that only considers clades at the top level when looking for differences.

**Comment by runjak** [comment]
Considering only clades where `level1 = 0`:
File | Entries
----- | -----
[1.txt](https://github.com/lingdb/CoBL/files/401640/1.txt) | 8
[2.txt](https://github.com/lingdb/CoBL/files/401639/2.txt) | 110

**Comment by runjak** [comment]
Some corrections/improvements on the last calculation (ignoring PIE):

File | Entries
----- | -----
[1.txt](https://github.com/lingdb/CoBL/files/401672/1.txt) | 7
[2.txt](https://github.com/lingdb/CoBL/files/401671/2.txt) | 105

**Comment by runjak** [comment]
Turns out the last computations didn't take the `Current` language list into account correctly. I've corrected the computations:

File | Entries
----- | -----
[1.txt](https://github.com/lingdb/CoBL/files/401695/1.txt) | 2
[2.txt](https://github.com/lingdb/CoBL/files/401694/2.txt) | 64

**Comment by PaulHeggarty** [comment]
"The latest results above are to be taken as the definitive ones.  For now."  :)

**Comment by runjak** [comment]
`Task 3` will be:

```
For each clade(level0!=0):
  print clade name
  print all cognate classes that don't have a root form and are only in this clade
```

**Comment by PaulHeggarty** [comment]
The corrections applied by Jakob to the lastest runs have cleaned up things a great deal.  

Task 1 is all but closed, with just two cognate classes needing to have a Root Form entered.

Task 2 is also looking much more manageable, with just 64 cognate classes to look at.

Task 3 will give separate statistics for each top-level clade, and the tasks of checking those up and entering `Root Form` and `Root Language` will be assigned to corresponding specialists for each clade.  Each task will be put into a corresponding issue, all with the same issue type label:  `Mono-Branch Cognate Sets:  [XXX Branch]`.

**Comment by cormacanderson** [comment]
Here is a sorted list for task 2, sorted by doublet, following the clade ordering on the languages references table 
[Cognate sets with 2 branches.txt](https://github.com/lingdb/CoBL/files/401775/Cognate.sets.with.2.branches.txt)

**Comment by runjak** [comment]
I've computed statistics per clade:

File | Entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/401753/Albanian.txt) | 126
[Anatolian.txt](https://github.com/lingdb/CoBL/files/401757/Anatolian.txt) | 27
[Armenian.txt](https://github.com/lingdb/CoBL/files/401752/Armenian.txt) | 131
[BalticEast.txt](https://github.com/lingdb/CoBL/files/401755/BalticEast.txt) | 21
[Baltic.txt](https://github.com/lingdb/CoBL/files/401751/Baltic.txt) | 117
[BaltoSlavic.txt](https://github.com/lingdb/CoBL/files/401754/BaltoSlavic.txt) | 401
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/401758/CelticBrythonic.txt) | 174
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/401760/CelticGoidelic.txt) | 150
[Celtic.txt](https://github.com/lingdb/CoBL/files/401756/Celtic.txt) | 298
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/401759/GermanicEast.txt) | 23
[GermanicNorth.txt](https://github.com/lingdb/CoBL/files/401761/GermanicNorth.txt) | 222
[GermanicNorthWest.txt](https://github.com/lingdb/CoBL/files/401764/GermanicNorthWest.txt) | 318
[Germanic.txt](https://github.com/lingdb/CoBL/files/401762/Germanic.txt) | 324
[GermanicWest.txt](https://github.com/lingdb/CoBL/files/401763/GermanicWest.txt) | 134
[Hellenic.txt](https://github.com/lingdb/CoBL/files/401767/Hellenic.txt) | 115
[Indic.txt](https://github.com/lingdb/CoBL/files/401766/Indic.txt) | 531
[IndoIranic.txt](https://github.com/lingdb/CoBL/files/401765/IndoIranic.txt) | 1306
[Iranic.txt](https://github.com/lingdb/CoBL/files/401769/Iranic.txt) | 764
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/401768/ItalicRoman.txt) | 212
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/401770/ItalicSabellic.txt) | 3
[Italic.txt](https://github.com/lingdb/CoBL/files/401773/Italic.txt) | 215
[Nuristani.txt](https://github.com/lingdb/CoBL/files/401771/Nuristani.txt) | 83
[Slavic.txt](https://github.com/lingdb/CoBL/files/401772/Slavic.txt) | 293
[Tocharian.txt](https://github.com/lingdb/CoBL/files/401774/Tocharian.txt) | 42

**Comment by PaulHeggarty** [comment]
Revision needed for these too, I’m afraid. 'GermanicNorthWest' should give only the set intersection of 'GermanicWest' and 'GermanicNorth', not the set union of those.  Same applies to all other nested clades.

**Comment by CasFre** [comment]
Thank you all for the new statistics.
As @nitsnatsnok is apparently taking care of the inner-Germanic root forms, I will start with the pronominal meanings now.

**Comment by runjak** [comment]
I've done the data per clade again this time filtering the data so that data for a clade does not contain data for sub-clades.

File | Entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/403504/Albanian.txt) | 126
[Anatolian.txt](https://github.com/lingdb/CoBL/files/403501/Anatolian.txt) | 27
[Armenian.txt](https://github.com/lingdb/CoBL/files/403500/Armenian.txt) | 131
[Baltic.txt](https://github.com/lingdb/CoBL/files/403502/Baltic.txt) | 107
[BalticEast.txt](https://github.com/lingdb/CoBL/files/403503/BalticEast.txt) | 21
[Celtic.txt](https://github.com/lingdb/CoBL/files/403505/Celtic.txt) | 10
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/403506/CelticBrythonic.txt) | 174
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/403507/CelticGoidelic.txt) | 150
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/403508/GermanicEast.txt) | 23
[GermanicNorth.txt](https://github.com/lingdb/CoBL/files/403509/GermanicNorth.txt) | 222
[GermanicWest.txt](https://github.com/lingdb/CoBL/files/403510/GermanicWest.txt) | 134
[Hellenic.txt](https://github.com/lingdb/CoBL/files/403511/Hellenic.txt) | 115
[Indic.txt](https://github.com/lingdb/CoBL/files/403512/Indic.txt) | 531
[Iranic.txt](https://github.com/lingdb/CoBL/files/403513/Iranic.txt) | 764
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/403515/ItalicRoman.txt) | 212
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/403514/ItalicSabellic.txt) | 3
[Nuristani.txt](https://github.com/lingdb/CoBL/files/403516/Nuristani.txt) | 83
[Slavic.txt](https://github.com/lingdb/CoBL/files/403517/Slavic.txt) | 293
[Tocharian.txt](https://github.com/lingdb/CoBL/files/403518/Tocharian.txt) | 42

**Comment by PaulHeggarty** [comment]
I’m afraid something is still not right with these statistics, and the files you sent me earlier.  Cognate set http://cobl.info/cognate/1011/ , for example, is still listed in both the CelticGoidelic and CelticBrythonic files, but those lists should be mutually exclusive.  Since it is in both branches, it should be in the overall Celtic list only, and not in either of the individual clade files.  There may be a problem with the erroneous Gaulish language causing a problem in this case, but the problem is more general.  There are also entries repeated in the GermanicWest and GermanicNorth files, when they should therefore be instead in GermanicNorthWest only (or indeed in Germanic only if they are also found in GermanicEast).

So far, I can use the mono-clade results only.  I presume they’re ok, although I note an inconsistency between your figures and the actual number of records.  In the Armenian file, you write 131 entries, but there are in fact 132.

Finally, in the export txt, could you write `cog. set` instad of `cog. class`, please?  Just for consistency with how we're calling these in the guidance notes.  Thanks.

**Comment by runjak** [comment]
Yes - I'll make sure to adjust my writing and check what I'm missing in the statistics.

**Comment by PaulHeggarty** [comment]
Another problem discovered:  the lists are still returning far too many entries, because they are still including lexemes marked as 'Not Swh'.   An example is the Hellenic list, which includes http://cobl.info/cognate/1518/, for example, even though that is marked as 'Not Swh'
Cormac identified an outstanding problem even with the 'Not Swh' filter on the `1-Meaning, All Cognate Sets` form as it still stands.  It is nonetheless returning cognate sets whose members are all marked at 'Not Swh' -- at least in some combinations of circumstances.  You’ll have to check with Cormac for the details.

**Comment by PaulHeggarty** [comment]
Because of the above problem with the statistics, I've temporarily closed all the issues I’ve been creating of the type:
'Anonymous' Cognate Sets only in Branch:  XXXX
We have to get the statistics problem sorted out first, or we'll be wasting experts' time to check over many cognate sets that are excluded in the first place.
Jakob, could you not create a pre-existing recordset of only lexemes not marked a 'Not Swadesh', and use that as the 'default' source recordset for these queries?  And only switch to the full recordset when the Not Swadesh filter button is switched off?

**Comment by CasFre** [comment]
Okay, so as far as inter-clade cognacy coding is concerned, I see the following things:

- Finish off task 2 according to @cormacanderson's file Cognate sets with 2 branches.txt.
- As soon as root forms + root languages is finished, another go on cognate set cross-checking would be advisable, i.e. checking whether all the lexemes assigned to that cognate class really belong there and whether further mergers with single clade sets are possible. The second task will of course to a great extent be done by branch experts that work through the branch specific lists.

**Comment by CasFre** [comment]
I would thus now start with task 2, suggesting that @RolandPooth concentrates on the Vedic language lists and on the cognacy coding for the Indic and Indo-Iranic languages, thus also doing his share of cognate set cross-checking.
Is that in line with overall priorities?

**Comment by runjak** [comment]
The miscalculation of entries for the statistics files is most likely due to me simply counting the lines per file but forgetting about the newline at the end of the file.
I've found some cases where I ignored the `not_swadesh_field` and fixed these.
Next will be:
* Double checking the clade logic
* Fetching a current copy of the database
* Computing updated statistics

**Comment by runjak** [comment]
New calculations with a recent database:

file | entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/406487/Albanian.txt) | 124
[Anatolian.txt](https://github.com/lingdb/CoBL/files/406488/Anatolian.txt) | 28
[Armenian.txt](https://github.com/lingdb/CoBL/files/406493/Armenian.txt) | 90
[Baltic.txt](https://github.com/lingdb/CoBL/files/406489/Baltic.txt) | 107
[BalticEast.txt](https://github.com/lingdb/CoBL/files/406491/BalticEast.txt) | 21
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/406492/CelticBrythonic.txt) | 152
[CelticContinental.txt](https://github.com/lingdb/CoBL/files/406490/CelticContinental.txt) | 11
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/406494/CelticGoidelic.txt) | 113
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/406496/GermanicEast.txt) | 22
[GermanicNorth.txt](https://github.com/lingdb/CoBL/files/406495/GermanicNorth.txt) | 185
[GermanicWest.txt](https://github.com/lingdb/CoBL/files/406497/GermanicWest.txt) | 130
[Hellenic.txt](https://github.com/lingdb/CoBL/files/406498/Hellenic.txt) | 72
[Indic.txt](https://github.com/lingdb/CoBL/files/406499/Indic.txt) | 504
[Iranic.txt](https://github.com/lingdb/CoBL/files/406501/Iranic.txt) | 737
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/406500/ItalicRoman.txt) | 191
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/406502/ItalicSabellic.txt) | 4
[Nuristani.txt](https://github.com/lingdb/CoBL/files/406503/Nuristani.txt) | 80
[Slavic.txt](https://github.com/lingdb/CoBL/files/406504/Slavic.txt) | 268
[Tocharian.txt](https://github.com/lingdb/CoBL/files/406505/Tocharian.txt) | 42

**Comment by cormacanderson** [comment]
Something is still wrong. The first entry in the Slavic list http://cobl.info/cognate/403/ also has an Albanian entry which is not marked as "Not Swh", so properly falls under Task 2, not Task 3.

**Comment by runjak** [comment]
Thanks for having a look, I'll investigate.

**Comment by runjak** [comment]
Wait!
So does this mean that you need task 3 to exclude the results from task 1 and 2?
Because this isn't happening at all at the moment because I'm working with them as separate tasks.

**Comment by runjak** [comment]
I've recomputed task3 this time excluding cases already found for task 2:

file | entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/406670/Albanian.txt) | 117
[Anatolian.txt](https://github.com/lingdb/CoBL/files/406671/Anatolian.txt) | 25
[Armenian.txt](https://github.com/lingdb/CoBL/files/406673/Armenian.txt) | 81
[BalticEast.txt](https://github.com/lingdb/CoBL/files/406674/BalticEast.txt) | 17
[Baltic.txt](https://github.com/lingdb/CoBL/files/406672/Baltic.txt) | 99
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/406678/CelticBrythonic.txt) | 149
[CelticContinental.txt](https://github.com/lingdb/CoBL/files/406675/CelticContinental.txt) | 10
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/406676/CelticGoidelic.txt) | 105
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/406679/GermanicEast.txt) | 17
[GermanicNorth.txt](https://github.com/lingdb/CoBL/files/406677/GermanicNorth.txt) | 175
[GermanicWest.txt](https://github.com/lingdb/CoBL/files/406680/GermanicWest.txt) | 120
[Hellenic.txt](https://github.com/lingdb/CoBL/files/406681/Hellenic.txt) | 72
[Indic.txt](https://github.com/lingdb/CoBL/files/406682/Indic.txt) | 502
[Iranic.txt](https://github.com/lingdb/CoBL/files/406683/Iranic.txt) | 725
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/406684/ItalicRoman.txt) | 175
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/406685/ItalicSabellic.txt) | 4
[Nuristani.txt](https://github.com/lingdb/CoBL/files/406686/Nuristani.txt) | 80
[Slavic.txt](https://github.com/lingdb/CoBL/files/406687/Slavic.txt) | 256
[Tocharian.txt](https://github.com/lingdb/CoBL/files/406688/Tocharian.txt) | 38

**Comment by cormacanderson** [comment]
Excellent - that looks much more manageable. Is task1 also excluded there?

**Comment by runjak** [comment]
If I'm not mistaken task1 should already be a subset of task2 and thus be excluded.

**Comment by cormacanderson** [comment]
Great, thanks. In that case then, I suppose we are good to go!

**Comment by runjak** [comment]
@PaulHeggarty spoke this issue in the car to Jena today.
The original tasks read `more than two branches` and `more than one branch`.
But it appears that it would be more helpful to have `task2` not include `task1` - is that correct? In that case I'll need to adjust the computations and produce the results again.

**Comment by runjak** [comment]
While talking about this with @PaulHeggarty in the car this morning I became aware that task2 still incorrectly includes task1 cases. I've made corrections for this and recalculated task1 and task2:

File | Entries
----- | -----
[1.txt](https://github.com/lingdb/CoBL/files/408614/1.txt) | 2
[2.txt](https://github.com/lingdb/CoBL/files/408613/2.txt) | 54

**Comment by CasFre** [comment]
Thank you all for that!
I should have thought of that in the first place when I opened this issue.

**Comment by CasFre** [comment]
Thank you all for that!
I should have thought of that in the first place when I opened this issue.

**Comment by runjak** [comment]
We just discovered an error in the statistics for task3 that I'll work on now:
Cases that appear in some subclades are placed in each subclade rather than the parent clade so far.

**Comment by runjak** [comment]
After speaking with @CasFre I've made some adjustments to task2.
The produced result is similar to the last one in the number of records:

File | Entries
----- | -----
[2.txt](https://github.com/lingdb/CoBL/files/408922/2.txt) | 54

@CasFre if something appears strange to you don't hesitate to alert me to it.

**Comment by runjak** [comment]
I've also adjusted task3:

File | Entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/408925/Albanian.txt) | 116
[Anatolian.txt](https://github.com/lingdb/CoBL/files/408924/Anatolian.txt) | 25
[Armenian.txt](https://github.com/lingdb/CoBL/files/408927/Armenian.txt) | 81
[Baltic.txt](https://github.com/lingdb/CoBL/files/408926/Baltic.txt) | 82
[BaltoSlavic.txt](https://github.com/lingdb/CoBL/files/408928/BaltoSlavic.txt) | 24
[Celtic.txt](https://github.com/lingdb/CoBL/files/408929/Celtic.txt) | 24
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/408930/CelticBrythonic.txt) | 127
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/408934/CelticGoidelic.txt) | 84
[Germanic.txt](https://github.com/lingdb/CoBL/files/408931/Germanic.txt) | 264
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/408935/GermanicEast.txt) | 1
[Hellenic.txt](https://github.com/lingdb/CoBL/files/408932/Hellenic.txt) | 44
[Indic.txt](https://github.com/lingdb/CoBL/files/408933/Indic.txt) | 447
[IndoIranic.txt](https://github.com/lingdb/CoBL/files/408936/IndoIranic.txt) | 59
[Iranic.txt](https://github.com/lingdb/CoBL/files/408939/Iranic.txt) | 671
[Italic.txt](https://github.com/lingdb/CoBL/files/408938/Italic.txt) | 1
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/408937/ItalicRoman.txt) | 174
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/408941/ItalicSabellic.txt) | 3
[Nuristani.txt](https://github.com/lingdb/CoBL/files/408940/Nuristani.txt) | 62
[Slavic.txt](https://github.com/lingdb/CoBL/files/408942/Slavic.txt) | 249
[Tocharian.txt](https://github.com/lingdb/CoBL/files/408943/Tocharian.txt) | 37

These are not currently sorted by meaning, which will be the next thing to happen.

**Comment by runjak** [comment]
I've implemented sorting the entries for task3 by meaning.
The results are:

File | Entries
----- | -----
[Albanian.txt](https://github.com/lingdb/CoBL/files/408976/Albanian.txt) | 117
[Anatolian.txt](https://github.com/lingdb/CoBL/files/408978/Anatolian.txt) | 25
[Armenian.txt](https://github.com/lingdb/CoBL/files/408979/Armenian.txt) | 81
[Baltic.txt](https://github.com/lingdb/CoBL/files/408980/Baltic.txt) | 82
[BaltoSlavic.txt](https://github.com/lingdb/CoBL/files/408977/BaltoSlavic.txt) | 24
[Celtic.txt](https://github.com/lingdb/CoBL/files/408981/Celtic.txt) | 24
[CelticBrythonic.txt](https://github.com/lingdb/CoBL/files/408984/CelticBrythonic.txt) | 127
[CelticGoidelic.txt](https://github.com/lingdb/CoBL/files/408986/CelticGoidelic.txt) | 85
[Germanic.txt](https://github.com/lingdb/CoBL/files/408985/Germanic.txt) | 259
[GermanicEast.txt](https://github.com/lingdb/CoBL/files/408983/GermanicEast.txt) | 1
[Hellenic.txt](https://github.com/lingdb/CoBL/files/408982/Hellenic.txt) | 44
[Indic.txt](https://github.com/lingdb/CoBL/files/408987/Indic.txt) | 447
[IndoIranic.txt](https://github.com/lingdb/CoBL/files/408988/IndoIranic.txt) | 59
[Iranic.txt](https://github.com/lingdb/CoBL/files/408989/Iranic.txt) | 672
[Italic.txt](https://github.com/lingdb/CoBL/files/408992/Italic.txt) | 1
[ItalicRoman.txt](https://github.com/lingdb/CoBL/files/408990/ItalicRoman.txt) | 174
[ItalicSabellic.txt](https://github.com/lingdb/CoBL/files/408991/ItalicSabellic.txt) | 3
[Nuristani.txt](https://github.com/lingdb/CoBL/files/408993/Nuristani.txt) | 62
[Slavic.txt](https://github.com/lingdb/CoBL/files/408994/Slavic.txt) | 249
[Tocharian.txt](https://github.com/lingdb/CoBL/files/408995/Tocharian.txt) | 37

**Comment by CasFre** [comment]
@runjak: I also splitt off some Iranian lexeme from a cognate class where otherwise only Albanian forms where entered: http://www.cobl.info/cognate/367/
For this cognate set the exact root form is unclear to me, and I did not put in the question mark at first.

As far as the statistics on Task 2 are concerned, I can't detect any difference between the last and the one-but-last lists.

**Comment by runjak** [comment]
I agree - there doesn't appear to be much of a difference.
Would you like me to rerun the statistics to reflect your change to cognate class 367?
Also: which question mark?

**Comment by leszekjocz** [comment]
Generally, there will be some changes here. E.g. I revealed that a so far 'anonymous' Slavic cognate class should be actually merged with a German cognate class and will not be 'anonymous' anymore. I'm sure we will reveal more cases like that. However, the lists are extremely useful no matter that.
Is the Slavic list ready? Should I update the issue for 'anonymous' Slavic cognate sets?

**Comment by CasFre** [comment]
cognate class 367 is already on the latest list for task 3 with respect to Albanian. So my change is reflected already. That's what I wanted to indicate.
I mean the question mark in the root ref. form field.

**Comment by runjak** [comment]
Ah, I think I found it, thanks :)
![screenshot_2016-08-09_15-31-55](https://cloud.githubusercontent.com/assets/5417642/17518025/6ddcddc8-5e46-11e6-897a-9c5e8cdd020b.png)
Interesting appearance.

**Comment by PaulHeggarty** [comment]
Regarding Leszek's comment:
_I've just revealed that a so far 'anonymous' Slavic cognate class should be actually merged with a German cognate class and will not be 'anonymous' anymore_
… there may be a need for a clarification here.  The name 'anonymous' only refers to the fact that there is no `Root ref. form` nor `Root ref. language`, entered, so that these cognate classes appear 'anonymous' in the drop-down box used for assigning a cognate class to a lexeme.
You seem to be referring here to a different concept, whis is that of a 'mono-branch' cognate set.  They are not the same thing, although the new issues being created for each branch are technically lists of all still anonymous mono-branch cognate sets in each branch.  As you work through them, and give them a `Root ref. form` and a `Root ref. language` entry, they would no longer be reported as 'anonymous' and so would not be identified if we run through these statistics again.  We will do so, in fact, from time to time, just to check on progress or whether any have still been missed.

**Comment by PaulHeggarty** [comment]
The title of this issue is about *statistics*, and I think is basically solved now, because the latest stage 3 outputs seem to be working as desired, at last.  So unless comments appear here soon, we should close this issue.

The debate on the policy to follow in entering `Root ref. form` and `Root ref. language` should I think now be moved to #274.  There are also other issues with the label 'Root Languages', so you can click on that label to see the list of them.

**Comment by CasFre** [comment]
Hello @runjak,
From our last chat on these statistics I understood that I should report to you if I still came across cases where a cognate class appeared in the task 2-list where excluded lexemes where taken into account in computing the number of branches involved.
I think that [2813](http://www.cobl.info/cognate/2813/) is a case in question, unless the Old Bretonic lexeme was marked as excluded only after the last compilation of the statistics.
